src/wavpack.c

```diff
@@ -0,0 +1,1032 @@
+) ;
+static inline int	wavpack_unpack_buffer_single_any
+(	SF_PRIVATE *psf, sf_count_t *total_packed_count, void *ptr, sf_count_t i_chunk, sf_count_t chunk_item_count, sf_count_t item_count, int32_t *buffer,
+	int itype, int otype, float vlim, int shl, int voffset
+) ;
+
+static sf_count_t	wavpack_unpack_buffer_any (SF_PRIVATE *psf, void *ptr, sf_count_t sample_count, int otype) ;
+static sf_count_t	wavpack_pack_buffer_any (SF_PRIVATE *psf, const void *ptr, sf_count_t sample_count, int itype) ;
+
+static int 		wavpack_enc_init (SF_PRIVATE *psf) ;
+static int 		wavpack_dec_init (SF_PRIVATE *psf) ;
+static int		wavpack_write_header (SF_PRIVATE *psf, int UNUSED (calc_length)) ;
+static int		wavpack_command (SF_PRIVATE *psf, int command, void *UNUSED (data), int UNUSED (datasize)) ;
+static int 		wavpack_close (SF_PRIVATE *psf) ;
+static sf_count_t	wavpack_seek (SF_PRIVATE *psf, int UNUSED (mode), sf_count_t offset) ;
+static int		wavpack_byterate (SF_PRIVATE *psf) ;
+
+static sf_count_t		wavpack_read_short (SF_PRIVATE *psf, short *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_read_int (SF_PRIVATE *psf, int *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_read_float (SF_PRIVATE *psf, float *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_read_double (SF_PRIVATE *psf, double *ptr, sf_count_t len) ;
+
+static sf_count_t		wavpack_write_short (SF_PRIVATE *psf, const short *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_write_int (SF_PRIVATE *psf, const int *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_write_float (SF_PRIVATE *psf, const float *ptr, sf_count_t len) ;
+static sf_count_t		wavpack_write_double (SF_PRIVATE *psf, const double *ptr, sf_count_t len) ;
+
+static int32_t	sf_wavpack_read_bytes_callback (void *client_data, void *data, int32_t bcount) ;
+static int32_t	sf_wavpack_write_bytes_callback (void *client_data, void *data, int32_t bcount) ;
+static int64_t	sf_wavpack_get_pos_callback (void *client_data) ;
+static int		sf_wavpack_set_pos_abs_callback (void *client_data, int64_t pos) ;
+static int		sf_wavpack_set_pos_rel_callback (void *client_data, int64_t delta, int mode) ;
+static int		sf_wavpack_push_back_byte_callback (void *id, int c) ;
+static int64_t	sf_wavpack_get_length_callback (void *client_data) ;
+static int		sf_wavpack_can_seek_callback (void *client_data) ;
+
+static WavpackStreamReader64	wavpack_reader_wrapper =
+{	sf_wavpack_read_bytes_callback,
+	sf_wavpack_write_bytes_callback,
+	sf_wavpack_get_pos_callback,
+	sf_wavpack_set_pos_abs_callback,
+	sf_wavpack_set_pos_rel_callback,
+	sf_wavpack_push_back_byte_callback,
+	sf_wavpack_get_length_callback,
+	sf_wavpack_can_seek_callback,
+	NULL,
+	NULL
+} ;
+
+
+/*------------------------------------------------------------------------------
+** Public function.
+*/
+
+int
+wavpack_open	(SF_PRIVATE *psf)
+{	// int		subformat ;
+	int		error = 0 ;
+
+	WAVPACK_PRIVATE* pwvpk = calloc (1, sizeof (WAVPACK_PRIVATE)) ;
```

Comment:
Add a null check and return SFE_MALLOC_FAILED if unsuccessful.

I know, you shouldn't really check for null, it's never handled correctly, and Linux's memory address space over-allocation means malloc is always successful. (You only find out it was a lie when using the memory causes a page-fault, there isn't any memory left and the OOM killer chooses you to die. In which case there's nothing to handle anyways.) But every other format does it, don't want to buck the trend.